src/serde_helpers.rs

```diff
@@ -0,0 +1,226 @@
+//! Collection of helper functions for serializing to and deserializing from BSON using Serde
+
+use std::{convert::TryFrom, result::Result};
+
+use serde::{de, ser, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{oid::ObjectId, Bson};
+
+/// Attempts to serialize a u32 as an i32. Errors if an exact conversion is not possible.
+pub fn serialize_u32_as_i32<S: Serializer>(val: &u32, serializer: S) -> Result<S::Ok, S::Error> {
```

Comment:
for the ones now included as part of modules, it would probably be nice to have aliases at the top level to match the format of these.

Reply:
Yeah you can via the `use` keyword. It's the same mechanism by which we move our exports around:

```rust
pub use some_module::some_fn as toplevel_some_fn;
```